checks/raw/pinned_dependencies.go

```diff
@@ -521,9 +524,11 @@ var validateDockerfilesPinning fileparser.DoWhileTrueOnFileContent = func(
 			// (1): name = <>@sha245:hash
 			// (2): name = XXX where XXX was pinned
 			pinned := pinnedAsNames[name]
+			// Record the asName.
 			if pinned || regex.MatchString(name) {
-				// Record the asName.
 				pinnedAsNames[asName] = true
+			} else {
+				pinnedAsNames[asName] = false
```

Comment:
this was a confusing line to review. Initially I didn't think it was needed as entries not found in the map return the zero value (false here).

But this has to do with build names that are re-used, e.g. `build3` is re-used in this other test case.
https://github.com/ossf/scorecard/blob/a22745007e0dbd6449bb03c26416325d1827006a/checks/raw/testdata/Dockerfile-pinned-as-without-hash#L26-L28